maybe use griffelib instead of griffe as dependency, and update to 2+

my ci and service occasionally crash with
```bash
/usr/local/lib/python3.13/site-packages/agents/tool.py:53: in <module>
    from .function_schema import DocstringStyle, function_schema
/usr/local/lib/python3.13/site-packages/agents/function_schema.py:10: in <module>
    from griffe import Docstring, DocstringSectionKind
/usr/local/lib/python3.13/site-packages/griffe/__init__.py:277: in <module>
    from griffe._internal.expressions import (
E   ImportError: cannot import name 'ExprInterpolation' from 'griffe._internal.expressions' (/usr/local/lib/python3.13/site-packages/griffe/_internal/expressions.py)
```
it may cause by 
```bash
uv pip list | grep griffe
Using Python 3.13.12 environment at: /usr/local
griffe                                       1.15.0
griffelib                                    2.0.0
```
i think they sometimes overwrite each other by installation order?

if using griffelib it may be resolved by uv

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -9,7 +9,7 @@ authors = [{ name = "OpenAI", email = "support@openai.com" }]
 dependencies = [
     "openai>=2.26.0,<3",
     "pydantic>=2.12.2, <3",
-    "griffe>=1.5.6, <2",
+    "griffelib>=2, <3",
     "typing-extensions>=4.12.2, <5",
     "requests>=2.0, <3",
     "types-requests>=2.0, <3",

--- a/src/agents/function_schema.py
+++ b/src/agents/function_schema.py
@@ -7,7 +7,8 @@ import re
 from dataclasses import dataclass
 from typing import Annotated, Any, Callable, Literal, get_args, get_origin, get_type_hints
 
-from griffe import Docstring, DocstringSectionKind
+# griffelib exposes the `griffe` package at runtime but currently does not ship typing markers.
+from griffe import Docstring, DocstringSectionKind  # type: ignore[import-untyped]
 from pydantic import BaseModel, Field, create_model
 from pydantic.fields import FieldInfo
 

